datasync/chaindatafetcher/repository.go

```diff
@@ -36,18 +36,13 @@ type Repository interface {
 	SetComponent(component interface{})
 }
 
-type retry func(blockchain.ChainEvent) error
-
-func retryFunc(f retry) retry {
-	return func(event blockchain.ChainEvent) error {
-		var err error
-		for {
-			if err = f(event); err == nil {
-				break
-			}
-			logger.Warn("retrying...", "blockNumber", event.Block.NumberU64())
+func retryFunc(f func(blockchain.ChainEvent) error) func(blockchain.ChainEvent) {
+	return func(event blockchain.ChainEvent) {
```

Comment:
Is there any way to stop this loop when user give kill signal?

Reply:
I moved the method under `ChainDataFetcher` and it returns when `stopCh` receives something.